orchagent/routeorch.cpp

```diff
@@ -2080,52 +2190,55 @@ bool RouteOrch::removeRoutePost(const RouteBulkContext& ctx)
         /* Delete Fine Grained nhg if the revmoved route pointed to it */
         m_fgNhgOrch->removeFgNhg(vrf_id, ipPrefix);
     }
-    else
+    /* Check that the next hop group is not owned by NhgOrch. */
+    else if (it_route->second.nhg_index.empty())
```

Comment:
please have this as `else if (!it_route->second.nhg_index.empty())` and retain the `else`

Reply:
Done